### ISSUE TITLE
ListView Items with visibility property throws binding warning if bound to null

### ISSUE THREAD

**Issue by Daxito** [NativeScript/NativeScript#2149]
### Did you verify this is a real problem by searching
Yes, nothing related found for NS 2.0.1

### Tell us about the problem
I have a list view, and this list view has a label (per item of course) that I hide using a converter, if the property bound to the Visibility is Null the console throws the warning:
**_"JS: Binding: Binding error while setting property visibility of Label(17)@file:///app/main-page.xml:9:17;: Error: Invalid value null for property visibility"_**
I think it was ok with NS 2.0.0 btw.

### Which platform(s) does your issue occur on?
Android

### Please provide the following version numbers that your issue occurs with:
- CLI: 2.0.1
- Cross-platform modules: 2.0.1
- Runtime(s): Android 2.0.0
- Plugin(s): None

### Please tell us how to recreate the issue in as much detail as possible.
You can recreate the issue by running the attached sample app, just run it and check the console log, it works but it gives the warning.

### Is there code involved? 
Yes, please download the attached sample app
[SampleApp2.zip](https://github.com/NativeScript/NativeScript/files/273333/SampleApp2.zip)

**Comment by tsonevn** [comment]
Hi @Daxito, 

Thank you for reporting this issue.

I reviewed your code and found that this is a real problem, related only with NS 2.0.1.
this problem will be scheduled for fixing.

Regards,
@tsonevn

**Comment by tsonevn** [comment]
Hi @Daxito,
Thank you for reporting your problem. 
I reviewed your sample project and we found a solution for your issue. Actually if you use ObservableArray in your main-view-model this warning will disappear and your application would work as expected. I will give you possible solution in below attached code snippets.

**main-page.xml**
```
<Page xmlns="http://schemas.nativescript.org/tns.xsd" navigatingTo="onNavigatingTo">
  <StackLayout>
    <Label text="Check Your Log ;-)" class="title"/>
    
    <ListView items="{{ source }}">
        <ListView.itemTemplate>
            <StackLayout>
                <Label text="{{text}}" textWrap="true" visibility="{{ isVisble ? 'visible' : 'collapsed' }}" />
                <Label text="{{text2}}" textWrap="true" visibility="{{ isVisble2 ? 'visible' : 'collapsed' }}" />
            </StackLayout>
        </ListView.itemTemplate>
    </ListView>
    
  </StackLayout>
</Page>
```

**main-view-model.js**
```
var Observable = require("data/observable").Observable;
var application = require("application");
var ObservableArray = require("data/observable-array").ObservableArray;


//Define global converter
var visibilityConverter = function (value) {         
    if ((value == 1) || (value == 2)) {
        return 'visible';
    } else {
        return 'collapsed';
    }
}
application.resources["visibilityConverter"] = visibilityConverter;

function createViewModel() {
    var viewModel = new Observable();
    var array= new ObservableArray();
    var someValue = ["test", null, 123];
    for( i=0; i<someValue.length; i++){
        if(someValue[i]!=null){
            array.push(new Observable({text:"text"+i, isVisble: true, text2: someValue[i],  isVisble2: true}));
        }
        else{
            console.log("null value")
            array.push(new Observable({text:"text1"+i, isVisble: true, text2: "no value",  isVisble2: false}));
        }
        
    }
    
    viewModel.source = array;

    return viewModel;
}

exports.createViewModel = createViewModel;
```

**main-page.js**
```
var createViewModel = require("./main-view-model").createViewModel;

function onNavigatingTo(args) {
    var page = args.object;
    page.bindingContext = createViewModel();
}
exports.onNavigatingTo = onNavigatingTo;
```



Let me know whether this helps.

**Comment by Daxito** [comment]
@tsonevn 
I assume you say I can use this as a workaround until you support this again.
I don't want to use a list of observables because I feel I would be adding extra overhead where I don't need it, the app still works so I think I can wait until fixed. 
Thanks!

**Comment by calebeaires** [comment]
👍  
Same issue, same problem!

**Comment by nsndeck** [comment]
The issue is fixed with commit #2299. 
Keep in mind that this fixes the issue with the `SampleApp2` project stated in the opening message. After fix - binding converters like `{{ sourceValue, sourceValue | converter }}` will be called even when the value of `sourceValue` is null or undefined. However if there is no converter or converter returns an invalid value for `visibility` property (different from "visible" or "collapse" error will be thrown), that error will be handled by binding infrastructure and will result as error warning in console. With 2.0.0 release we print all errors in console in order to help resolving common problems with binding, css styling and so on. Generally any error message is a potential problem in the running application.
